editoast/src/views/layers/layer.rs

```diff
@@ -91,6 +96,62 @@ pub async fn layer_view(
     }))
 }
 
+/// Gets mvt tile from the cache if possible, otherwise gets data fro the data base and caches it in redis
+#[get(
+    "/tile/<layer_slug>/<view_slug>/<z>/<x>/<y>?<infra>",
+    format = "application/octet-stream"
+)]
+#[allow(clippy::too_many_arguments)]
+pub async fn cache_and_get_mvt_tile<'a>(
+    layer_slug: &str,
+    view_slug: &str,
+    z: u64,
+    x: u64,
+    y: u64,
+    infra: i64,
+    map_layers: &State<MapLayers>,
+    conn: DBConnection,
+    pool: &RedisPool,
+) -> ApiResult<Vec<u8>> {
+    let layer = match map_layers.layers.get(layer_slug) {
+        Some(layer) => layer,
+        None => return Err(LayersError::new_layer_not_found(layer_slug, map_layers).into()),
+    };
+    let view = match layer.views.get(view_slug) {
+        Some(view) => view,
+        None => return Err(LayersError::new_view_not_found(view_slug, layer).into()),
+    };
+    let cache_key = get_cache_tile_key(
+        &get_view_cache_prefix(layer_slug, infra, view_slug),
+        &Tile { x, y, z },
+    );
+    let cached_value: Vec<u8> = get_or_default(pool, &cache_key).await;
```

Comment:
We should call a `get` function that returns an `Option` and the check if the option is `Some` or `None`. We can avoid to recompute an empty tile.